eta/core/features.py

```diff
@@ -189,16 +189,26 @@ class CanFeaturize(object):
     managing and using a Featurizer instance.
     '''
 
-    def __init__(self, featurizer=None, hint_featurize=False):
-        '''Initialize CanFeaturize instance.
+    def __init__(self, featurizer=None, force_featurize=False):
+        '''Initializes a CanFeaturize instance.
 
         Args:
             featurizer: the actual featurizer to use when needed
-            hint_featurize: when True, this will force any input to the
-            decorator featurize_if_needed to always featurize.
+            force_featurize: whether to force any input to the
+                `featurize_if_needed` decorator to be featurized. By default,
+                this is False
         '''
         self.featurizer = featurizer
-        self.hint_featurize = hint_featurize
+        self.force_featurize = force_featurize
+
+    def __enter__(self):
```

Comment:
Although this looks attractive, I do worry about whether this will get confusing for downstream developers as classes extending `CanFeaturize` will likely expose similar environment functionality for themselves.

Reply:
Upon second thought, I think this is the wrong place to implement this. I'm moving it to `Classifier`